mawabot/cogs/messages/core.py

```diff
@@ -49,43 +49,50 @@ def __init__(self, bot):
     async def hit(self, ctx, *, content: str = None):
         ''' Sends the given message and then immediately deletes it '''
 
-        fut = ctx.message.delete()
+        await asyncio.gather(
+            self._hit(ctx, content),
+            ctx.message.delete(),
+        )
+
+    @staticmethod
+    async def _hit(ctx, content):
         if content is not None:
             await ctx.send(content=content, delete_after=0)
-        await fut
 
     @commands.command()
     async def delay(self, ctx, seconds: float, *, content: str):
         ''' Sends the given message after the specified number of seconds '''
 
         logger.info(f'Queued up delayed message for {seconds} seconds from now')
-        await ctx.message.delete()
-        await asyncio.sleep(seconds)
+        await asyncio.gather(
+            asyncio.sleep(seconds),
+            ctx.message.delete(),
```

Comment:
Shouldn't this be the other way around? So the command is deleted before waiting instead of after.

Reply:
The order doesn't matter, it runs them all in parallel.